futures-select-macro/Cargo.toml

```diff
@@ -0,0 +1,25 @@
+[package]
+name = "futures-select-macro-preview"
+edition = "2018"
+version = "0.3.0-alpha.10"
+authors = ["Taylor Cramer <cramertj@google.com>"]
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/rust-lang-nursery/futures-rs"
+homepage = "https://rust-lang-nursery.github.io/futures-rs"
+documentation = "https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.10/futures_select"
+description = """
+The `select!` macro for waiting on multiple different `Future`s at once and handling the first one to complete.
+"""
+
+[lib]
+name = "futures_select_macro"
+proc-macro = true
+
+[features]
+std = []
```

Comment:
At some point I figured we could use this to turn off `rand` support, though I've left that unimplemented for now. Perhaps we'd require the user to provide a link-time function that could be used to shuffle arrays?

Reply:
Yep, something like that sounds possible. I was thinking I might take a look at getting it working on `nostd`, then realised it wouldn’t work without `await!` support.

Reply:
Right, `await!` is the big blocker for getting no-std.